contracts/Collector.sol

```diff
@@ -73,12 +83,23 @@ contract Collector is ICollector{
             totalShares = totalShares + _partners[i].share;
 
         require(totalShares == 100, "total shares must add up to 100%");
-
         _;
     }
 
     modifier onlyOwner(){
         require(msg.sender == owner, "can only call from owner");
         _;
     }
+
+    modifier noRemainders(){
+        uint balance = token.balanceOf(address(this));
+        require(balance < partners.length, "balance is not a remainder");
+        if(balance == 0) {
+            return;
```

Comment:
I'm not sure about this line. We'll probably still need to use `_`. We need to check because I don't remember exactly.